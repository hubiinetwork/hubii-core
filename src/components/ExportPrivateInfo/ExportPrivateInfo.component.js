import React from 'react';
import PropTypes from 'prop-types';
import CopyToClipboard from 'react-copy-to-clipboard';
import Notification from '../Notification';

import {
  Text,
  TextPrimary,
  Wrapper,
  ParentDiv,
  SecondaryHeader,
  StyledButton,
  StyledIcon,
  PrimaryHeader,
} from './ExportPrivateInfo.style';

/**
 * ExportPrivateInfo
 */
export default class ExportPrivateInfo extends React.PureComponent {
  constructor(props) {
    super(props);
    this.showNotification = this.showNotification.bind(this);
  }

  showNotification(type) {
    const success = true;
    const message = `${type} copied to clipboard.`;
    Notification(success, message);
  }

  render() {
    const { mnemonic, privateKey, onExit } = this.props;
    return (
      <Wrapper>
        <Text>Never share this information with anyone</Text>
        <Text>Always keep a physical backup in a safe location</Text>
        <TextPrimary>
          {
              mnemonic ?

                <div>
                  <SecondaryHeader>
                    Mnemonic
                    <CopyToClipboard text={mnemonic} >
                      <StyledIcon
                        type="primary"
                        shape="circle"
                        icon="copy"
                        size={'small'}
                        onClick={() => this.showNotification('Mnemonic')}
                        id="mnemonic"
                      />
                    </CopyToClipboard>
                  </SecondaryHeader>
                  {mnemonic}
                </div> :
                <div>
                  <PrimaryHeader>
                    Mnemonic
                  </PrimaryHeader>
                  This wallet was imported using a private key therefore does not have a mnemonic to export
                </div>
            }
          <div>
            <SecondaryHeader>
              Private key
            <CopyToClipboard text={privateKey} >
              <StyledIcon
                type="primary"
                shape="circle"
                icon="copy"
                size={'small'}
                onClick={() => this.showNotification('Private key')}
                id="privateKey"
              />
            </CopyToClipboard>
            </SecondaryHeader>
            {privateKey}
          </div>
        </TextPrimary>
        <ParentDiv>
          <StyledButton type="primary" onClick={onExit} id="exit">
            Close
          </StyledButton>
        </ParentDiv>
      </Wrapper>
    );
  }
}

ExportPrivateInfo.propTypes = {
  /**
   * Function to perform action when exit button is clicked
   */
  onExit: PropTypes.func.isRequired,
  /**
   * Wallet mnemonic
   */
  mnemonic: PropTypes.string,
  /**
   * Wallet private key
   */
  privateKey: PropTypes.string,
};
