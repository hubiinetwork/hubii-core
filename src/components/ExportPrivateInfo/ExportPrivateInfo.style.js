import styled from 'styled-components';

import Button from 'components/ui/Button';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
export const Text = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: large;
  margin-bottom: 0.5rem;
`;

export const SecondaryHeader = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: medium;
  margin: 0.5rem 0 0.5rem 2rem;
`;

export const PrimaryHeader = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: medium;
  margin: 0.5rem 0 0.5rem 0rem;
`;

export const TextPrimary = styled.div`
  color: ${({ theme }) => theme.palette.light};
  margin-top: 1rem;
  text-align: center;
`;
export const StyledIcon = styled(Button)`
  margin-left: 0.5rem;
  color: ${({ theme }) => theme.palette.secondary1};
  background: ${({ theme }) => theme.palette.secondary8};
  border-color: ${({ theme }) => theme.palette.secondary8};
  &:hover {
      color: ${({ theme }) => theme.palette.info};
      background: ${({ theme }) => theme.palette.secondary8} !important;
      border-color: ${({ theme }) => theme.palette.secondary8} !important;
    }
`;

export const StyledButton = styled(Button)`
  min-width: 10.71rem;
  border-width: 0.14rem;
  padding: 0.5rem 1rem;
  margin: 1rem;
  color: ${({ theme }) => theme.palette.light};
`;

export const ParentDiv = styled.div`
  margin-top: 1.43rem;
  display: flex;
  justify-content: center;
`;
