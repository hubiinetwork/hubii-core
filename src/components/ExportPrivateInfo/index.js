export { default } from './ExportPrivateInfo.component';
