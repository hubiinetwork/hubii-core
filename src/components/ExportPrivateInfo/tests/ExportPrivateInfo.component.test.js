import React from 'react';
import { shallow } from 'enzyme';
import ExportPrivateInfo from 'components/ExportPrivateInfo';

describe('<ExportPrivateInfo/>', () => {
  const wallet = {
    name: 'Wallet1',
    address: '0x3123123',
    privateKey: 'privateKey',
    mnemonic: 'the ball in my hand',
  };
  const onExit = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ExportPrivateInfo
        name={wallet.name}
        address={wallet.address}
        onExit={onExit}
        privateKey={wallet.privateKey}
        mnemonic={wallet.mnemonic}
      />
    );
  });

  it('should render <ExportPrivateInfo/> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('when the exit button is clicked the onExit function is run', () => {
    const button = wrapper.find('#exit');
    button.simulate('click');
    expect(onExit).toHaveBeenCalledTimes(1);
  });
  describe('showNotification function', () => {
    it('should run the showNoftification function when the Icon to copy a wallet name is clicked', () => {
      const button = wrapper.find('#name');
      const spy = jest.spyOn(wrapper.instance(), 'showNotification');
      button.simulate('click');
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('should run the showNoftification function when the Icon to copy a wallet address is clicked', () => {
      const button = wrapper.find('#address');
      const spy = jest.spyOn(wrapper.instance(), 'showNotification');
      button.simulate('click');
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('should run the showNoftification function when the Icon to copy a wallet mnemonic is clicked', () => {
      const button = wrapper.find('#mnemonic');
      const spy = jest.spyOn(wrapper.instance(), 'showNotification');
      button.simulate('click');
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('should run the showNoftification function when the Icon to copy a wallet private key is clicked', () => {
      const button = wrapper.find('#privateKey');
      const spy = jest.spyOn(wrapper.instance(), 'showNotification');
      button.simulate('click');
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
