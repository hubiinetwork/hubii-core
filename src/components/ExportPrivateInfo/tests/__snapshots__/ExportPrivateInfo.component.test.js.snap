// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportPrivateInfo/> should render <ExportPrivateInfo/> correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExportPrivateInfo
    address="0x3123123"
    mnemonic="the ball in my hand"
    name="Wallet1"
    onExit={[Function]}
    privateKey="privateKey"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ExportPrivateInfo.style__Text>
          Never share this information with anyone
</ExportPrivateInfo.style__Text>,
        <ExportPrivateInfo.style__Text>
          Always keep a physical backup in a safe location
</ExportPrivateInfo.style__Text>,
        <ExportPrivateInfo.style__TextPrimary>
          <div>
                    <ExportPrivateInfo.style__SecondaryHeader>
                              Mnemonic
                              <CopyToClipboard
                                        onCopy={undefined}
                                        options={undefined}
                                        text="the ball in my hand"
                              >
                                        <ExportPrivateInfo.style__StyledIcon
                                                  icon="copy"
                                                  id="mnemonic"
                                                  onClick={[Function]}
                                                  shape="circle"
                                                  size="small"
                                                  type="primary"
                                        />
                              </CopyToClipboard>
                    </ExportPrivateInfo.style__SecondaryHeader>
                    the ball in my hand
          </div>
          <div>
                    <ExportPrivateInfo.style__SecondaryHeader>
                              Private Key
                              <CopyToClipboard
                                        onCopy={undefined}
                                        options={undefined}
                                        text="privateKey"
                              >
                                        <ExportPrivateInfo.style__StyledIcon
                                                  icon="copy"
                                                  id="privateKey"
                                                  onClick={[Function]}
                                                  shape="circle"
                                                  size="small"
                                                  type="primary"
                                        />
                              </CopyToClipboard>
                    </ExportPrivateInfo.style__SecondaryHeader>
                    privateKey
          </div>
</ExportPrivateInfo.style__TextPrimary>,
        <ExportPrivateInfo.style__ParentDiv>
          <ExportPrivateInfo.style__StyledButton
                    id="exit"
                    onClick={[Function]}
                    type="primary"
          >
                    Close
          </ExportPrivateInfo.style__StyledButton>
</ExportPrivateInfo.style__ParentDiv>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Never share this information with anyone",
        },
        "ref": null,
        "rendered": "Never share this information with anyone",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Always keep a physical backup in a safe location",
        },
        "ref": null,
        "rendered": "Always keep a physical backup in a safe location",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              <ExportPrivateInfo.style__SecondaryHeader>
                            Mnemonic
                            <CopyToClipboard
                                          onCopy={undefined}
                                          options={undefined}
                                          text="the ball in my hand"
                            >
                                          <ExportPrivateInfo.style__StyledIcon
                                                        icon="copy"
                                                        id="mnemonic"
                                                        onClick={[Function]}
                                                        shape="circle"
                                                        size="small"
                                                        type="primary"
                                          />
                            </CopyToClipboard>
              </ExportPrivateInfo.style__SecondaryHeader>
              the ball in my hand
</div>,
            <div>
              <ExportPrivateInfo.style__SecondaryHeader>
                            Private Key
                            <CopyToClipboard
                                          onCopy={undefined}
                                          options={undefined}
                                          text="privateKey"
                            >
                                          <ExportPrivateInfo.style__StyledIcon
                                                        icon="copy"
                                                        id="privateKey"
                                                        onClick={[Function]}
                                                        shape="circle"
                                                        size="small"
                                                        type="primary"
                                          />
                            </CopyToClipboard>
              </ExportPrivateInfo.style__SecondaryHeader>
              privateKey
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ExportPrivateInfo.style__SecondaryHeader>
                  Mnemonic
                  <CopyToClipboard
                                    onCopy={undefined}
                                    options={undefined}
                                    text="the ball in my hand"
                  >
                                    <ExportPrivateInfo.style__StyledIcon
                                                      icon="copy"
                                                      id="mnemonic"
                                                      onClick={[Function]}
                                                      shape="circle"
                                                      size="small"
                                                      type="primary"
                                    />
                  </CopyToClipboard>
</ExportPrivateInfo.style__SecondaryHeader>,
                "the ball in my hand",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Mnemonic",
                    <CopyToClipboard
                      onCopy={undefined}
                      options={undefined}
                      text="the ball in my hand"
>
                      <ExportPrivateInfo.style__StyledIcon
                                            icon="copy"
                                            id="mnemonic"
                                            onClick={[Function]}
                                            shape="circle"
                                            size="small"
                                            type="primary"
                      />
</CopyToClipboard>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Mnemonic",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <ExportPrivateInfo.style__StyledIcon
                        icon="copy"
                        id="mnemonic"
                        onClick={[Function]}
                        shape="circle"
                        size="small"
                        type="primary"
/>,
                      "onCopy": undefined,
                      "options": undefined,
                      "text": "the ball in my hand",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "icon": "copy",
                        "id": "mnemonic",
                        "onClick": [Function],
                        "shape": "circle",
                        "size": "small",
                        "type": "primary",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "the ball in my hand",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ExportPrivateInfo.style__SecondaryHeader>
                  Private Key
                  <CopyToClipboard
                                    onCopy={undefined}
                                    options={undefined}
                                    text="privateKey"
                  >
                                    <ExportPrivateInfo.style__StyledIcon
                                                      icon="copy"
                                                      id="privateKey"
                                                      onClick={[Function]}
                                                      shape="circle"
                                                      size="small"
                                                      type="primary"
                                    />
                  </CopyToClipboard>
</ExportPrivateInfo.style__SecondaryHeader>,
                "privateKey",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Private Key",
                    <CopyToClipboard
                      onCopy={undefined}
                      options={undefined}
                      text="privateKey"
>
                      <ExportPrivateInfo.style__StyledIcon
                                            icon="copy"
                                            id="privateKey"
                                            onClick={[Function]}
                                            shape="circle"
                                            size="small"
                                            type="primary"
                      />
</CopyToClipboard>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Private Key",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <ExportPrivateInfo.style__StyledIcon
                        icon="copy"
                        id="privateKey"
                        onClick={[Function]}
                        shape="circle"
                        size="small"
                        type="primary"
/>,
                      "onCopy": undefined,
                      "options": undefined,
                      "text": "privateKey",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "icon": "copy",
                        "id": "privateKey",
                        "onClick": [Function],
                        "shape": "circle",
                        "size": "small",
                        "type": "primary",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "privateKey",
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ExportPrivateInfo.style__StyledButton
            id="exit"
            onClick={[Function]}
            type="primary"
>
            Close
</ExportPrivateInfo.style__StyledButton>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Close",
            "id": "exit",
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": "Close",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ExportPrivateInfo.style__Text>
            Never share this information with anyone
</ExportPrivateInfo.style__Text>,
          <ExportPrivateInfo.style__Text>
            Always keep a physical backup in a safe location
</ExportPrivateInfo.style__Text>,
          <ExportPrivateInfo.style__TextPrimary>
            <div>
                        <ExportPrivateInfo.style__SecondaryHeader>
                                    Mnemonic
                                    <CopyToClipboard
                                                onCopy={undefined}
                                                options={undefined}
                                                text="the ball in my hand"
                                    >
                                                <ExportPrivateInfo.style__StyledIcon
                                                            icon="copy"
                                                            id="mnemonic"
                                                            onClick={[Function]}
                                                            shape="circle"
                                                            size="small"
                                                            type="primary"
                                                />
                                    </CopyToClipboard>
                        </ExportPrivateInfo.style__SecondaryHeader>
                        the ball in my hand
            </div>
            <div>
                        <ExportPrivateInfo.style__SecondaryHeader>
                                    Private Key
                                    <CopyToClipboard
                                                onCopy={undefined}
                                                options={undefined}
                                                text="privateKey"
                                    >
                                                <ExportPrivateInfo.style__StyledIcon
                                                            icon="copy"
                                                            id="privateKey"
                                                            onClick={[Function]}
                                                            shape="circle"
                                                            size="small"
                                                            type="primary"
                                                />
                                    </CopyToClipboard>
                        </ExportPrivateInfo.style__SecondaryHeader>
                        privateKey
            </div>
</ExportPrivateInfo.style__TextPrimary>,
          <ExportPrivateInfo.style__ParentDiv>
            <ExportPrivateInfo.style__StyledButton
                        id="exit"
                        onClick={[Function]}
                        type="primary"
            >
                        Close
            </ExportPrivateInfo.style__StyledButton>
</ExportPrivateInfo.style__ParentDiv>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Never share this information with anyone",
          },
          "ref": null,
          "rendered": "Never share this information with anyone",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Always keep a physical backup in a safe location",
          },
          "ref": null,
          "rendered": "Always keep a physical backup in a safe location",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                <ExportPrivateInfo.style__SecondaryHeader>
                                Mnemonic
                                <CopyToClipboard
                                                onCopy={undefined}
                                                options={undefined}
                                                text="the ball in my hand"
                                >
                                                <ExportPrivateInfo.style__StyledIcon
                                                                icon="copy"
                                                                id="mnemonic"
                                                                onClick={[Function]}
                                                                shape="circle"
                                                                size="small"
                                                                type="primary"
                                                />
                                </CopyToClipboard>
                </ExportPrivateInfo.style__SecondaryHeader>
                the ball in my hand
</div>,
              <div>
                <ExportPrivateInfo.style__SecondaryHeader>
                                Private Key
                                <CopyToClipboard
                                                onCopy={undefined}
                                                options={undefined}
                                                text="privateKey"
                                >
                                                <ExportPrivateInfo.style__StyledIcon
                                                                icon="copy"
                                                                id="privateKey"
                                                                onClick={[Function]}
                                                                shape="circle"
                                                                size="small"
                                                                type="primary"
                                                />
                                </CopyToClipboard>
                </ExportPrivateInfo.style__SecondaryHeader>
                privateKey
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ExportPrivateInfo.style__SecondaryHeader>
                    Mnemonic
                    <CopyToClipboard
                                        onCopy={undefined}
                                        options={undefined}
                                        text="the ball in my hand"
                    >
                                        <ExportPrivateInfo.style__StyledIcon
                                                            icon="copy"
                                                            id="mnemonic"
                                                            onClick={[Function]}
                                                            shape="circle"
                                                            size="small"
                                                            type="primary"
                                        />
                    </CopyToClipboard>
</ExportPrivateInfo.style__SecondaryHeader>,
                  "the ball in my hand",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Mnemonic",
                      <CopyToClipboard
                        onCopy={undefined}
                        options={undefined}
                        text="the ball in my hand"
>
                        <ExportPrivateInfo.style__StyledIcon
                                                icon="copy"
                                                id="mnemonic"
                                                onClick={[Function]}
                                                shape="circle"
                                                size="small"
                                                type="primary"
                        />
</CopyToClipboard>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Mnemonic",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <ExportPrivateInfo.style__StyledIcon
                          icon="copy"
                          id="mnemonic"
                          onClick={[Function]}
                          shape="circle"
                          size="small"
                          type="primary"
/>,
                        "onCopy": undefined,
                        "options": undefined,
                        "text": "the ball in my hand",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "icon": "copy",
                          "id": "mnemonic",
                          "onClick": [Function],
                          "shape": "circle",
                          "size": "small",
                          "type": "primary",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "the ball in my hand",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ExportPrivateInfo.style__SecondaryHeader>
                    Private Key
                    <CopyToClipboard
                                        onCopy={undefined}
                                        options={undefined}
                                        text="privateKey"
                    >
                                        <ExportPrivateInfo.style__StyledIcon
                                                            icon="copy"
                                                            id="privateKey"
                                                            onClick={[Function]}
                                                            shape="circle"
                                                            size="small"
                                                            type="primary"
                                        />
                    </CopyToClipboard>
</ExportPrivateInfo.style__SecondaryHeader>,
                  "privateKey",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Private Key",
                      <CopyToClipboard
                        onCopy={undefined}
                        options={undefined}
                        text="privateKey"
>
                        <ExportPrivateInfo.style__StyledIcon
                                                icon="copy"
                                                id="privateKey"
                                                onClick={[Function]}
                                                shape="circle"
                                                size="small"
                                                type="primary"
                        />
</CopyToClipboard>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Private Key",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <ExportPrivateInfo.style__StyledIcon
                          icon="copy"
                          id="privateKey"
                          onClick={[Function]}
                          shape="circle"
                          size="small"
                          type="primary"
/>,
                        "onCopy": undefined,
                        "options": undefined,
                        "text": "privateKey",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "icon": "copy",
                          "id": "privateKey",
                          "onClick": [Function],
                          "shape": "circle",
                          "size": "small",
                          "type": "primary",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "privateKey",
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ExportPrivateInfo.style__StyledButton
              id="exit"
              onClick={[Function]}
              type="primary"
>
              Close
</ExportPrivateInfo.style__StyledButton>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "id": "exit",
              "onClick": [Function],
              "type": "primary",
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
